lib/Segments/WooCommerce.php

```diff
@@ -0,0 +1,279 @@
+<?php
+namespace MailPoet\Segments;
+
+use MailPoet\Models\ModelValidator;
+use MailPoet\Models\Subscriber;
+use MailPoet\Models\Segment;
+use MailPoet\Models\SubscriberSegment;
+use MailPoet\Newsletter\Scheduler\Scheduler;
+use MailPoet\Settings\SettingsController;
+use MailPoet\Subscribers\Source;
+
+if(!defined('ABSPATH')) exit;
+
+class WooCommerce {
+  /** @var SettingsController */
+  private $settings;
+
+  function __construct(SettingsController $settings) {
+    $this->settings = $settings;
+  }
+
+  function synchronizeRegisteredCustomer($wp_user_id, $current_filter = null) {
+    if(!$current_filter && !$this->settings->get('enable_wc_hooks_testing')) {
+      return false; // temporarily disable hooks (except for testing)
+    }
+
+    $wc_segment = Segment::getWooCommerceSegment();
+
+    if($wc_segment === false) return;
+
+    $current_filter = $current_filter ?: current_filter();
+    switch($current_filter) {
+      case 'woocommerce_delete_customer':
+        // subscriber should be already deleted in WP users sync
+        $this->unsubscribeUsersFromSegment(); // remove leftover association
+        break;
+      case 'woocommerce_new_customer':
+        $new_customer = true;
+      case 'woocommerce_update_customer':
+      default:
+        $wp_user = \get_userdata($wp_user_id);
```

Comment:
We shouldn't use WP functions directly. We should use the \MailPoet\WP\Functions class.

Reply:
Fixed.